kani-driver/src/main.rs

```diff
@@ -66,6 +66,11 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
 
     let mut failed_harnesses: Vec<&HarnessMetadata> = Vec::new();
 
+    if ctx.args.print_harness_metadata {
```

Comment:
I would suggest that you move this code to be inside `collect_kani_metadata`. You can print the metadata there.